pkg/controllers/clusterresourceplacement/clusterresourceplacement_controller.go

```diff
@@ -21,6 +22,58 @@ func (r *Reconciler) Reconcile(_ context.Context, _ controller.QueueKey) (ctrl.R
 	return ctrl.Result{}, nil
 }
 
+func (r *Reconciler) handleDelete(ctx context.Context, crp *fleetv1beta1.ClusterResourcePlacement) (ctrl.Result, error) {
+	crpKObj := klog.KObj(crp)
+	if !controllerutil.ContainsFinalizer(crp, fleetv1beta1.ClusterResourcePlacementCleanupFinalizer) {
+		klog.V(4).InfoS("clusterResourcePlacement is being deleted", "clusterResourcePlacement", crpKObj)
+		return ctrl.Result{}, nil
+	}
+	klog.V(2).InfoS("Removing clusterResourcePlacement", "clusterResourcePlacement", crpKObj)
```

Comment:
This should be `Removing snapshots` I think.

Reply:
And maybe we should add separate logs for policy snapshots and resource snapshots?